sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/MetricReaderFactory.java

```diff
@@ -34,7 +36,7 @@ static MetricReaderFactory getInstance() {
   }
 
   @Override
-  public MetricReader create(
+  public MetricReaderAndCardinalityLimits create(
       MetricReaderModel model, SpiHelper spiHelper, List<Closeable> closeables) {
     PeriodicMetricReaderModel periodicModel = model.getPeriodic();
     if (periodicModel != null) {
```

Comment:
nit: I like factoring expressive methods for these if blocks -- like `if(periodicModel != null){ return buildPeriodicModelReaderWithLimits(...) }` unless the param list is too unwieldy.

Reply:
I don't disagree. In a well factored version of this, there's a `Factory` implementation for `PeriodicMetricReaderModel` and `PullMetricReaderFactory`. You can take this pattern too far and have a factory for each and every model type, however trivial, and it ends up making it harder to read.

I ended up making the change here because this method is starting to get too large to easily reason about, but going to punt on applying it everywhere because I do think its a bit of a case by case thing.